src/main/java/com/hubspot/jackson/datatype/protobuf/builtin/deserializers/MessageDeserializer.java

```diff
@@ -109,7 +109,8 @@ protected void populate(
 
     Map<String, FieldDescriptor> fieldLookup = new HashMap<>();
     for (FieldDescriptor field : descriptor.getFields()) {
-      fieldLookup.put(namingStrategy.translate(field.getName()), field);
+      String fieldName = field.toProto().hasJsonName() ? field.getJsonName() : field.getName();
+      fieldLookup.put(namingStrategy.translate(fieldName), field);
```

Comment:
Seems like this will apply the naming strategy to the `json_name` value, do you know if that matches the proto3 `JsonFormat` behavior?

Reply:
Seems like `JsonFormat` just writes the json_name exactly without any transformation